lua/plenary/async/async.lua

```diff
@@ -50,13 +51,13 @@ do
     __mode = "k",
   })
 
-  add_leaf_function = function(async_func)
+  add_leaf_function = function(async_func, argc)
     assert(_PlenaryLeafTable[async_func] == nil, "Async function should not already be in the table")
-    _PlenaryLeafTable[async_func] = true
+    _PlenaryLeafTable[async_func] = argc
   end
 
-  function M.is_leaf_function(async_func)
-    return _PlenaryLeafTable[async_func] ~= nil
+  function M.get_leaf_function(async_func)
```

Comment:
I think `get_leaf_fucntion_argc` is more clear, then we can also add `is_leaf_function` based off of that

Reply:
Personally I find having two functions pretty redundant especially given the triviality and also since it is only for internal logic. It's pretty idiomatic in Lua to just check for nil. Either way I don't feel too strong about it.

Reply:
sure that makes sense